engine/src/main/java/org/hibernate/validator/internal/metadata/location/ConstraintLocation.java

```diff
@@ -44,7 +44,11 @@ static ConstraintLocation forField(Field field) {
 	}
 
 	static ConstraintLocation forGetter(Method getter) {
-		return new GetterConstraintLocation( getter );
+		return new GetterConstraintLocation( getter , getter.getDeclaringClass() );
+	}
+
+	static ConstraintLocation forGetterOfClass(Method getter, Class<?> declaringClass) {
```

Comment:
I would put `declaringClass` first and we definitely need a comment here.

And I would simply name it `forGetter()` too.

Reply:
I initially overloaded the `forGetter()` method and it raised a lot of test failures as I think there are tests using reflection to inspect this interface. I'll rename as suggested and comment here with more details of failing tests and look for your direction. Thanks.

Reply:
Mmmmh, that's weird, I wouldn't expect any issue here. But I haven't tried.